base/parse.jl

```diff
@@ -339,4 +340,7 @@ tryparse_internal(::Type{T}, s::AbstractString, startpos::Int, endpos::Int, rais
     tryparse_internal(T, s, startpos, endpos, 10, raise)
 
 parse(::Type{T}, s::AbstractString) where T<:Union{Real,Complex} =
-    tryparse_internal(T, s, firstindex(s), lastindex(s), true)
+    tryparse_internal(T, s, firstindex(s), lastindex(s), true)::T
+
+tryparse(T::Type{Complex{S}}, s::AbstractString) where S<:Real =
+    tryparse_internal(T, s, firstindex(s), lastindex(s), false)::Union{T, Nothing}
```

Comment:
If you use `tryparse_internal(Complex{S}, ...)` can you remove the typeassert?

Reply:
Unfortunately no, as `tryparse_internal` has `Any` inferred type for `Compex` - also see my comment below I made to the second commit (or I do not understand something).

Reply:
I know :(.
Simply I could not locate the reason why `tryparse_internal` was not type stable (as it seems it should when simply analyzing its code).

I will have a more detailed look at it and let you know if I am able to fix it.